player/external_files.c

```diff
@@ -110,36 +110,52 @@ static int compare_sub_priority(const void *a, const void *b)
     return strcoll(s1->fname, s2->fname);
 }
 
-static struct bstr guess_lang_from_filename(struct bstr name, int *fn_start)
+static struct bstr guess_lang_from_filename(struct bstr name, int *lang_start)
 {
     if (name.len < 2)
         return (struct bstr){NULL, 0};
 
-    int n = 0;
+    int lang_length = 0;
     int i = name.len - 1;
 
-    char thing = '.';
+    char delimiter = '.';
     if (name.start[i] == ')') {
-        thing = '(';
+        delimiter = '(';
         i--;
     }
     if (name.start[i] == ']') {
-        thing = '[';
+        delimiter = '[';
         i--;
     }
 
     while (i >= 0 && mp_isalpha(name.start[i])) {
-        n++;
-        if (n > 3)
-            return (struct bstr){NULL, 0};
+        lang_length++;
         i--;
     }
 
-    if (n < 2 || i == 0 || name.start[i] != thing)
+    if (lang_length < 2 || lang_length > 6 || (name.start[i] != '-' && lang_length > 3))
+        return (struct bstr){NULL, 0};
+
+    // Detect locales with a region, like en-US.
+    if (name.start[i] == '-') {
```

Comment:
BCP 47 allows multiple subtags. Like `gsw-u-sd-chzh` this should be supported. As I see currently only one would be added.